### PR TITLE
Support parameter filters on project list

### PR DESCRIPTION
**Pull request by ryanannatuvasquez**
## Description
- Support parameter filters on project list

### CODE REVIEW

File: memsource/api.py [hunk 1]
```diff
@@ -316,8 +316,8 @@ def create(self, name: str, source_lang: str, target_langs: Union[List[str], Tup
             'domain': domain,
         })['id']
 
-    def list(self):
-        return [models.Project(project) for project in self._post('project/list', {})]
+    def list(self, params: dict={}):
```
Comment (narusemotoki) [inline memsource/api.py]: Shoudn't this method takes `name` instead of `params`? Difficult to know what parameter we can pass.

Comment (ryanannatuvasquez) [reply memsource/api.py]: @narusemotoki however its more flexible if we pass `dict` as parameter to support available params from memsource
https://wiki.memsource.com/wiki/Project_API_v3#List_Projects

Comment (narusemotoki) [reply memsource/api.py]: I don't think it is easy to use. If you really want's to support all parameters, how about `name: str. **extra_parameters`?

Comment (kelvintaywl) [reply memsource/api.py]: just a suggestion, but maybe:

```python
def list(self, **query):
     return [
          models.Project(project)
          for project in self._post('project/list', query)
     ]
```

Comment (kelvintaywl) [reply memsource/api.py]: [small opinion]

I think it reads better if we use `**extra_parameters` (or perhaps `**query`) so the usage now looks like:

```python
def list(self, **query):
    return [models.Project(project) for project in self._post('project/list', query)]
```
```diff
# usage 
instead of:
- self.project.list(params={'name': 'foo project'})
we now have
+ self.project.list(name='foo project'}
```

🍜

**Comment by ryanannatuvasquez** [comment]
PTAL @narusemotoki 🙇  and also @kelvintaywl :bowtie: 🍺 🍻

**Pull request merged**